first try

wielembe opened the pull request:
no to próba nr 1

wielembe committed:
first try

no to próba nr 1

KamilMr commented on models/contacts.js:
```diff
@@ -1,19 +1,115 @@
+        } else {
+            contacts.splice(index, 1);
+            const newContacts = JSON.stringify(contacts);
+            await writeFile(contactsPath, newContacts, (err) => {
```
tu już nie dodajesz call back bo jest await
```suggestion
            await writeFile(contactsPath, newContacts)
```

wielembe replied:
ok

KamilMr commented on models/contacts.js:
```diff
@@ -1,19 +1,115 @@
+    }
+};
+
+const addContact = async (body) => {
```
W funkcji addContact brakuje sprawdzenia, czy kontakt o podanym id lub email już istnieje. Może to prowadzić do duplikatów w danych.

wielembe replied:
zagubiłam się. co zmieniam i kombinuję - to psuję. zakładałam, że jest strasznie małe prawdopodobieństwo na powtórzenie id. z mailem jeszcze walczę, ale na wszelki - proszę o wskazówkę, bo coś mi tu nie idzie

KamilMr commented on models/contacts.js:
```diff
@@ -1,19 +1,115 @@
+};
 
-const removeContact = async (contactId) => {}
+const getContactById = async (contactId) => {
```
Warto dodać walidację id przed przeszukaniem listy kontaktów. To może pomóc w obsłudze niepoprawnych lub nieistniejących id

wielembe replied:
jak wyżej. proszę  o wskazówkę.

KamilMr commented on models/contacts.js:
W funkcjach addContact i updateContact mieszasz użycie async/await z callbackami. Dla spójności i czytelności lepiej trzymać się jednego stylu.

KamilMr commented on routes/api/contacts.js:
```diff
@@ -1,25 +1,126 @@
+    try {
+        const body = {
+            id: v4(),
+            name: req.query.name,
```
W metodach POST i PUT wykorzystujesz req.query, co jest nietypowe. Zazwyczaj dane z formularza przesyła się za pomocą req.body. Upewnij się, że używasz odpowiedniego middleware'a do parsowania ciała żądania, np. express.json().

wielembe replied:
ok

KamilMr commented on routes/api/contacts.js:
```diff
@@ -1,25 +1,126 @@
-})
+router.put("/:contactId", async (req, res, next) => {
+    try {
+        const body = {
+            name: req.query.name,
+            email: req.query.email,
+            phone: req.query.phone,
+        };
```
Użyj req.body

wielembe replied:
ok

KamilMr reviewed (changes requested):
Podsumowując, twój kod jest dobrze zorganizowany i czytelny. Przedstawione sugestie mogą pomóc w poprawie niektórych aspektów obsługi błędów i danych wejściowych.

wielembe committed:
changes

wielembe committed:
próbuję zrobić walidację id

wielembe committed:
add contact update

dodałam logikę, że kontakt nie jest dodawany jeśli mail już istnieje. ale coś mi nie styka ze statusem (pokazuje success)

KamilMr commented on routes/api/contacts.js:
```diff
@@ -1,25 +1,126 @@
+            phone: req.body.phone,
+        };
+        const result = await updateContact(req.params.contactId, body);
+        if (result && result.status !== 400 && result !== 400) {
```
Możesz postarać się ujednolicić błędy jakie mogą przyjść z funkcji podrzędnych. Bo na przykład w tym sprawdzeniu result może być albo może być undefined. Ponadto może mieć status albo i nie. Ponadto liczba 400 może być przypisana do status albo do result. Dużo jest zmiennych które mogą przyjść. Struktura przechodzenia błędu powinna być stała żebyś mogła w łatwy sposób przewidywać co przyjdzie na przykład z updteContact. Z mojej perspektywy im mniej komplikacji tym lepiej. :)

KamilMr commented on models/contacts.js:
```diff
@@ -1,19 +1,126 @@
+        const mailIndex = contacts.findIndex(
+            (contact) => contact.email === body.email
+        );
+        if (!mailIndex) {
```
```suggestion
        if (mailIndex === -1) {
```

KamilMr reviewed (approved):
Prowadź sobie te zmiany i na końcu zrób merge :)

wielembe committed:
Update models/contacts.js

Co-authored-by: Kamil Mrowka <68347303+KamilMr@users.noreply.github.com>
